components/storage/blockdevice/COMPONENT_SPIF/SPIFBlockDevice.cpp

```diff
@@ -540,7 +550,7 @@ spif_bd_error SPIFBlockDevice::_spi_send_program_command(int prog_inst, const vo
 
 spif_bd_error SPIFBlockDevice::_spi_send_erase_command(int erase_inst, bd_addr_t addr, bd_size_t size)
 {
-    addr = (((int)addr) & 0x00FFF000);
+    addr = (((int)addr) & 0xFFFFF000);
```

Comment:
A magic number - please replace with a constant with a descriptive name

Reply:
Any change required? To be honest I don't think this binary and-operation is needed because the datasheet of the memory I use states:

> SUBSECTOR ERASE / SECTOR ERASE Sets the selected subsector or sector bits to FFh. Any address within the subsector is valid for entry. Each address bit is latched in during the rising edge of the clock.

But I didn't want to make the call to remove this line as I only can test with one flash chip and maybe it is there for an undocumented reason.

I needed to extend the bitmask with bit 24-31 set to one to be able to erase beyond 16Mbyte.

Reply:
Fine by me.